### ISSUE TITLE
Autocrafting Speed Upgrades not working

### ISSUE THREAD

**Issue by llloyd4** [raoulvdberge/refinedstorage#909]
#### Issue description:

So, like I did with all the processors and making the silicone and stuff I put down a Solderer, put down a chest one block above it, a Crafter to the back of that, and put in Ender I/O Item Conduits with filters on the input side to filter the components to the correct sides. Left is bottom, top is middle, and right is top. It worked on the processors but it's not working for the speed upgrade. Nothing is leaving the chest, nothing is going in, it just sits there with all the components in the input chest. Rechecking one of the processors, left is the imprinted silicone, right, that goes into the bottom slot, top is redstone, right as that goes in the middle, and right is the printed processor which goes in the top slot, so correct again. I autocraft the processor and it works. So I am puzzled. 

#### What happens:

The redstone, sugar, and upgrade modules sit in the input chest and don't go into the Solderer.

#### What you expected to happen:

The items to go into the Solderer and it autocraft Speed upgrades.

#### Steps to reproduce:
1. Put down Solderer. 
2. Put down chest one block above Solderer.
3. Using Ender I/O item conduits connect chest to left, top, and right sides of the Solderer, set connection to chest to Extract, and Always Active, set each side to Insert, put in Basic Filters, set left to whitelist redstone, top to whitelist upgrade module, and right to whitelist sugar. 
4. Manually place sugar, redstone, and an upgrade module into the chest to test.

#### Version (Make sure you are on the latest version before reporting):

- Minecraft: 1.10.2
- Forge: 12.18.3.2185
- Refined Storage: 1.2.17

Does this issue occur on a server? [yes/no] No idea, I am playing in a local world.

#### If a (crash)log is relevant for this issue, link it here:

No crashes but here are some screenshots so you can see my settings and setup

Overall Setup - [URL=http://s15.photobucket.com/user/llloyd4/media/2017-01-23_11.18.55_zpsbnrfvpaf.png.html][IMG]http://i15.photobucket.com/albums/a395/llloyd4/2017-01-23_11.18.55_zpsbnrfvpaf.png[/IMG][/URL]

Contents of the Chest - [URL=http://s15.photobucket.com/user/llloyd4/media/2017-01-23_10.57.55_zpsi48vfv21.png.html][IMG]http://i15.photobucket.com/albums/a395/llloyd4/2017-01-23_10.57.55_zpsi48vfv21.png[/IMG][/URL]

Extract settings - [URL=http://s15.photobucket.com/user/llloyd4/media/2017-01-23_10.58.08_zpsk0z6ikjl.png.html][IMG]http://i15.photobucket.com/albums/a395/llloyd4/2017-01-23_10.58.08_zpsk0z6ikjl.png[/IMG][/URL]

Left side (bottom slot) - [URL=http://s15.photobucket.com/user/llloyd4/media/2017-01-23_10.58.38_zps66jj7upy.png.html][IMG]http://i15.photobucket.com/albums/a395/llloyd4/2017-01-23_10.58.38_zps66jj7upy.png[/IMG][/URL]

Top side (middle slot) - [URL=http://s15.photobucket.com/user/llloyd4/media/2017-01-23_10.59.00_zpsbhwp60fo.png.html][IMG]http://i15.photobucket.com/albums/a395/llloyd4/2017-01-23_10.59.00_zpsbhwp60fo.png[/IMG][/URL]

Right side (top slot) - [URL=http://s15.photobucket.com/user/llloyd4/media/2017-01-23_10.59.09_zpsnnlwusix.png.html][IMG]http://i15.photobucket.com/albums/a395/llloyd4/2017-01-23_10.59.09_zpsnnlwusix.png[/IMG][/URL]

**Comment by Rudolpher** [comment]
Not able to replicate with same version. Have you verified all filters are correct, nothing is in the solderer and broken/replaced the extract on chest? There is an odd issue with cables that cause them to stop functioning from time to time.

**Comment by llloyd4** [comment]
Ok, that fixed the first problem. Now I am getting redstone in the middle slot instead of the bottom one. I really wish the patterns would be unified, redstone only goes into the bottom or middle in all recipes then all I would need is a Crafter and it redstone couldn't go into either slot. >.< Would probably mean reconfiguring things but, eh, once done it wouldn't need changing ever again and be a lot simpler.

**Comment by way2muchnoise** [comment]
They were unified as of #832. I think those are in 1.2.17, could be 1.2.18.

**Comment by llloyd4** [comment]
I am on 1.2.17 and was avoiding 1.2.18 due to another bug report someone else made about things being broken.

**Comment by raoulvdberge** [comment]
Try to remake the solderer patterns, their order was changed.

**Comment by raoulvdberge** [comment]
Also, the solderer is no longer sided. They auto insert in the right slot, it doesn't matter which side you insert them.

**Comment by llloyd4** [comment]
It didn't and I am running ver 17, I'll snapshot the gui showing sugar /
Redstone / upgrade that it can do but is wrong. Do that when I get home.

On Jan 23, 2017 12:17 PM, "Raoul" <notifications@github.com> wrote:

Also, the solderer is no longer sided. They auto insert in the right slot,
it doesn't matter which side you insert them.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/raoulvdberge/refinedstorage/issues/909#issuecomment-274604199>,
or mute the thread

**Comment by llloyd4** [comment]
Ok, so, it allows me to do this : 

[URL=http://s15.photobucket.com/user/llloyd4/media/2017-01-23_20.31.43_zps3aeno4ag.png.html][IMG]http://i15.photobucket.com/albums/a395/llloyd4/2017-01-23_20.31.43_zps3aeno4ag.png[/IMG][/URL]

This : 

[URL=http://s15.photobucket.com/user/llloyd4/media/2017-01-23_20.32.20_zpsxku9nrps.png.html][IMG]http://i15.photobucket.com/albums/a395/llloyd4/2017-01-23_20.32.20_zpsxku9nrps.png[/IMG][/URL]

Even this:

[URL=http://s15.photobucket.com/user/llloyd4/media/2017-01-23_20.32.30_zpsvrmsdf2x.png.html][IMG]http://i15.photobucket.com/albums/a395/llloyd4/2017-01-23_20.32.30_zpsvrmsdf2x.png[/IMG][/URL]

And as there is no ISides any more I can't even force it to behave, depending on what goes into the chest first is what it'll do so I could try to auto craft speed  upgrades and get the first one, I might get the right one, I might get one of the other ones, it's completely random now if it'll behave as, as you can see, it isn't only going into the correct slots. And this is 17.

**Comment by llloyd4** [comment]
Hi! I don't want to sound pushy, was curious if anyone saw my screen shots showing stuff going into the wrong slots on a Solderer ? I've not seen, in the last two days, any sort of response? Should I post this as a new Issue? Just curious. O.o

**Comment by raoulvdberge** [comment]
Update RS and remake your solderer patterns.

**Comment by llloyd4** [comment]
It is updated to the last working version unless there is a 19 now out? I remember in another someone said 18 the autocrafting was broke and downgrading fixed it. O.o So I have been holding off for 19.

**Comment by raoulvdberge** [comment]
Stop spamming this issue until you've tried .18.

**Comment by llloyd4** [comment]
Ok. Will get back to you.

**Comment by llloyd4** [comment]
Just got home from work, upgraded to .18, and the above screen shots are still perfectly valid, I can get sugar and two upgrades, sugar and two redstone, and sugar then redstone then and upgrade into the solderer in the wrong slots. And for giggles I also broke and replaced the solderer and same difference, still getting the wrong things in the wrong slots.

**Comment by raoulvdberge** [comment]
Did you recreate the patterns?

**Comment by llloyd4** [comment]
I manually crafted, no pattern needed. I had red stone, two upgrades, and sugar in my inventory. I clicked on the solderer to open the GUI. I shift clicked sugar first, then redstone, then the upgrade and they went in, in that order, sugar on top, redstone in the middle, upgrade on the bottom. Seeing that I took out the upgrade and redstone and tried shift clicking two upgrades, both went in to the middle and bottom slots. Then I took those upgrades out and to finish the same experiment as above I shift clicked on two individual pieces of redstone and, like above, they went into the middle and bottom slots.
